### PR TITLE
feat: update users-mgmt put endpoint

### PR DESCRIPTION
**Pull request by danielgent**
:rocket: nice. this actually fixes the PUT endpoint

[Screencast from 2023-09-10 22-20-36.webm](https://github.com/sesamyab/auth/assets/8496063/6d4f0d33-87ec-425f-8b1f-ca92f2ca3977)

### CODE REVIEW

File: src/routes/tsoa/users-mgmt.ts [hunk 1]
```diff
@@ -179,9 +179,12 @@ export class UsersMgmtController extends Controller {
     });
   }
 
-  @Put("users")
+  // this is the one used by RA for updating
+  @Put("users/{userId}/identities")
```
Comment (danielgent) [inline src/routes/tsoa/users-mgmt.ts:183]: I think this was the issue I was complaining about? :thinking: 

Is it our router that returns a 204 when it should be a 404?   Is there a security reason for this?

File: src/routes/tsoa/users-mgmt.ts [hunk 2]
```diff
@@ -179,14 +179,33 @@ export class UsersMgmtController extends Controller {
     });
   }
 
-  @Put("users")
```
Comment (danielgent) [inline src/routes/tsoa/users-mgmt.ts:182]: lol. No wonder this wasn't working :smile: 

Would be cool to get a 404 back when we haven't configured a route :thinking:  I"m sure there's a reason why not

File: src/routes/tsoa/users-mgmt.ts [hunk 3]
```diff
@@ -179,24 +179,22 @@ export class UsersMgmtController extends Controller {
...
+  @Put("users/{userId}")
   public async putUser(
     @Request() request: RequestWithContext,
+    @Path("userId") userId: string,
     @Header("tenant-id") tenantId: string,
     @Body()
-    user: Omit<User, "tenantId" | "createdAt" | "modifiedAt"> &
-      Partial<Pick<User, "createdAt" | "modifiedAt">>,
+    profile: Profile,
```
Comment (danielgent) [inline src/routes/tsoa/users-mgmt.ts:188]: this is the issue. react-admin is sending up an object with snake-cased e.g. `created_at`

:shrug:   Probably worth me understanding where the inconsistencies are coming from *but* this fixes it for now :smile:

Comment (markusahlstrand) [reply src/routes/tsoa/users-mgmt.ts:188]: It's a good question. So far we have been using camel casing for the management api and snake casing for the auth endpoints. I'm not sure if auth0 uses snake for both?

React-admin is a bit funny with the data it's passing to the server on an update.. So it always passes the entire entity, but we don't want to accept the createdAt and modifiedAt to be set from the client, so if you look in the dataProvider you can see that we're removing these before sending the data to the server. So the reason you get this issue is that there's a casing issue and I think that's what we should fix. We probably don't want to allow the createdAt or modifiedAt to be passed from the client.

File: src/routes/tsoa/users-mgmt.ts (hunk 2 above)
Comment (danielgent) [reply src/routes/tsoa/users-mgmt.ts:182]: I don't think it's urgent :smile: 

I had a quick look at the Hono docs. Looks great. Very similar to your cloudworker router so _shouldn't_ be a big job to swap it out :crossed_fingers:

File: src/routes/tsoa/users-mgmt.ts (hunk 3 above)
Comment (danielgent) [reply src/routes/tsoa/users-mgmt.ts:188]: @markusahlstrand https://app.shortcut.com/sesamy/story/4362/investigate-casing-used-by-auth0-camel-vs-snake ticket done here!

*so* I'm assuming _yes we're going with snake_case everywhere_


_BUT_ I will address your feedback now to not send up any timestamps up to the server

I'm guessing the auth0 management API doesn't? I'll check now :eyes:

Comment (danielgent) [reply src/routes/tsoa/users-mgmt.ts:188]: Indeed, looks that way :+1:

File: src/routes/tsoa/users-mgmt.ts (hunk 2 above)
Comment (markusahlstrand) [reply src/routes/tsoa/users-mgmt.ts:182]: It's a big nasty handlebars template :) Started looking at it but never got through :)

File: src/routes/tsoa/users-mgmt.ts (hunk 3 above)
Comment (danielgent) [reply src/routes/tsoa/users-mgmt.ts]: This is addressed on my `react-admin` PR here https://github.com/sesamyab/auth-admin/pull/36/files#diff-e42b04044d9f52c4cf5b23aaa94f2d676c252da5bfa4e0c8741080b798d14985R15

I'll put all these other PRs on hold though until we've decided on the casing on Auth2 -> https://github.com/sesamyab/auth/pull/133

**Comment by danielgent** [comment]
Is this still needed? I think we fixed this on the other PRs changing everything

I'm trying to get auth-admin working now, so let's see

**Pull request merged**